src/main.ts

```diff
@@ -31,34 +36,43 @@ export async function activate(context: vscode.ExtensionContext): Promise<{ init
     const reporter = new TelemetryReporter(extensionId, extensionVersion, aiKey);
 
     util.setExtensionPath(extension.extensionPath);
-    
+
+    const eventStream = new EventStream();
+
     let dotnetChannel = vscode.window.createOutputChannel('.NET');
     let dotnetChannelObserver = new DotNetChannelObserver(dotnetChannel);
     let dotnetLoggerObserver = new DotnetLoggerObserver(dotnetChannel);
+    eventStream.subscribe(dotnetChannelObserver.post);
+    eventStream.subscribe(dotnetLoggerObserver.post);
 
     let csharpChannel = vscode.window.createOutputChannel('C#');
     let csharpchannelObserver = new CsharpChannelObserver(csharpChannel);
     let csharpLogObserver = new CsharpLoggerObserver(csharpChannel);
+    eventStream.subscribe(csharpchannelObserver.post);
+    eventStream.subscribe(csharpLogObserver.post);
 
     let omnisharpChannel = vscode.window.createOutputChannel('OmniSharp Log');
```

Comment:
Should `omnisharpChannel` (and its observers) be owned by `OmniSharp` in https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/omnisharp/extension.ts?

Reply:
@DustinCampbell Right now we are trying to initialise all the observers in the main.ts so that it is that component that does the wiring up. But if you still feel that we need to move those observers into extension.ts we can do that.